test/rules/invalid_case_patterns_test.dart

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(InvalidCasePatternsTest);
+  });
+}
+
+@reflectiveTest
+class InvalidCasePatternsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'invalid_case_patterns';
+
+  test_binaryExpression() async {
+    await assertDiagnostics(r'''
+void f(Object o) {
+  switch (o) {
+    case 1 + 2:
+  }
+}
+''', [
+      lint(43, 5),
+    ]);
+  }
+
+  test_binaryExpression_logic() async {
+    await assertDiagnostics(r'''
+f(bool b) {
+  switch (b) {
+    case true && false:
+  }
+}
+''', [
+      lint(36, 13),
+    ]);
+  }
+
+  test_conditionalExpression() async {
+    await assertDiagnostics(r'''
+void f(Object o) {
+  switch (o) {
+    case true ? 1 : 2: 
+  }
+}
+''', [
+      lint(43, 12),
+      error(HintCode.DEAD_CODE, 54, 1),
+    ]);
+  }
+
+  test_constConstructorCall() async {
```

Comment:
We also need a test in which there's an explicit `const` keyword, because I think that one is valid in 3.0.

Reply:
👍  Added.  Thanks.